src/dataflow/src/source/postgres.rs

```diff
@@ -197,26 +197,47 @@ impl PostgresSourceReader {
             .parse()
             .or_else(|_| Err(anyhow!("invalid lsn"))));
         for info in publication_tables {
-            // TODO(petrosagg): use a COPY statement here for more efficient network transfer
-            let data = client
-                .simple_query(&format!(
-                    "SELECT * FROM {:?}.{:?}",
-                    info.namespace, info.name
-                ))
+            let relation_id: Datum = (info.rel_id as i32).into();
+            let reader = client
+                .copy_out_simple(
+                    format!(
+                        "COPY {:?}.{:?} TO STDOUT (FORMAT TEXT, DELIMITER '\t')",
+                        info.namespace, info.name
+                    )
+                    .as_str(),
+                )
                 .await?;
-            for msg in data {
-                if let SimpleQueryMessage::Row(row) = msg {
+
+            pin_mut!(reader);
+            while let Some(Ok(b)) = reader.next().await {
```

Comment:
processing each row as it comes in will allow us to avoid blocking until the entire dataset is buffered before beginning parsing, which is the current behavior. This additionally lets us avoid copying the whole dataset twice, since we can process the bytes from the stream directly